extension/lib/mergeCartsOnLogin.js

```diff
@@ -0,0 +1,16 @@
+'use strict'
+
+const ShopgateCartPipeline = require('./shopgate/CartExtensionPipeline')
+
+module.exports = async (context, input) => {
+  const anonymousPipeline = ShopgateCartPipeline.createForDevice(context)
+  const loggedInPipeline = ShopgateCartPipeline.create(context)
+
+  const itemsToAdd = (await anonymousPipeline.get()).cartItems.map(item => ({
```

Comment:
We should have error handling implemented like in other steps. Error handling in general should be improved but it doesn't need to be part of this PR. We should just be consistent.
After adding error handling, practically logging, you'll have more cases for unit test.